### PR TITLE
Add initial Trace API plugin docs to nodeos

### LINKED ISSUE #8808: Create nodeos documentation for Trace API plugin
Provide initial server-side nodeos documentation to configure the new Trace API plugin.

### PR DESCRIPTION
**Pull request by lparisc**
## Change Description

Resolves #8808 

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [x] Documentation Additions

### CODE REVIEW

File: docs/01_nodeos/03_plugins/trace_api_plugin/index.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                                        Failure to specify this option when
+                                        there are no trace-rpc-abi
+                                        configurations will result in an Error.
+                                        This option is mutually exclusive with
+                                        trace-rpc-api
+```
+
+## Dependencies
+
+* [`chain_plugin`](../chain_plugin/index.md)
```
Comment (heifner) [inline docs/01_nodeos/03_plugins/trace_api_plugin/index.md]: This is not exactly true. `chain_plugin` is a dependency but so is `http_plugin` neither of which have to be specified on the command line or in config.ini

File: docs/01_nodeos/03_plugins/trace_api_plugin/index.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
+# trace_history_plugin
+
+## Description
+
+The `trace_api_plugin` provides a consumer-focused long-term API for retrieving blocks containing retired actions and related metadata.
+
+## Usage
```
Comment (heifner) [inline docs/01_nodeos/03_plugins/trace_api_plugin/index.md]: Somewhere we should indicate the new http endpoint `/v1/trace_api/get_block`. See trace_api_plugin PR for more info.

**Comment by heifner** [comment]
Docs for #8802

File: docs/01_nodeos/03_plugins/trace_api_plugin/index.md (hunk 1 above)
Comment (lparisc) [reply docs/01_nodeos/03_plugins/trace_api_plugin/index.md]: Ok I will also add the http_plugin as a dependency (we also do it for most other plugins regardless of whether they need to be loaded explicitly). But I will add a comment in the "Load Dependency Examples" section below that "the chain_plugin and http_plugin will be loaded with default settings if not specified on the command line or config.ini". If ok please resolve conversation.

File: docs/01_nodeos/03_plugins/trace_api_plugin/index.md (hunk 2 above)
Comment (lparisc) [reply docs/01_nodeos/03_plugins/trace_api_plugin/index.md]: Got it, will add here. It will also be covered in the RPC API reference (PR coming shortly).

Comment (heifner) [reply docs/01_nodeos/03_plugins/trace_api_plugin/index.md]: Ah, ok if in there then that is fine. Just thought we needed it somewhere.

**Pull request merged**